syntax = "proto2";

package devtools_crosstool_autofdo;


// Next Available: 11.
message PropellerOptions {
  // binary file name.
  optional string binary_name = 1;

  // perf.data file name.
  repeated string perf_names = 2;

  // propeller cluster information output file name.
  optional string cluster_out_name = 3;

  // propeller symbol order output file name.
  optional string symbol_order_out_name = 4;

  // profiled binary name; if not empty, use this as file name to match MMAP
  // events in perf.data file, can be "".
  optional string profiled_binary_name = 5;

  // ignore the build id; if set, use binary_name or profiled_binary_name to
  // match MMAP events. Otherwise, use buildid (if exists) to match.
  optional bool ignore_build_id = 6;

  // Internal options used by unittest to keep frontend data.
  optional bool keep_frontend_intermediate_data = 7;

  // Parameters used to calculate the Extended TSP score for layout.
  optional PropellerCodeLayoutParameters code_layout_params = 8;

  // Include extra information such as per-function layout scores in the
  // propeller cluster file.
  optional bool verbose_cluster_output = 9 [default = false];

  optional bool dump_static_cfgs = 10 [default = false];
}

// Next Available: 6.
message PropellerCodeLayoutParameters {
  optional uint32 fallthrough_weight = 1 [default = 10];
  optional uint32 forward_jump_weight = 2 [default = 1];
  optional uint32 backward_jump_weight = 3 [default = 1];
  optional uint32 forward_jump_distance = 4 [default = 1500];
  optional uint32 backward_jump_distance = 5 [default = 1500];
}
